# Lab | Hypothesis Testing


### Instructions

1. It is assumed that the mean systolic blood pressure is `μ = 120 mm Hg`. In the Honolulu Heart Study, a sample of `n = 100` people had an average systolic blood pressure of 130.1 mm Hg with a standard deviation of 21.21 mm Hg. Is the group significantly different (with respect to systolic blood pressure!) from the regular population?

   - Set up the hypothesis test.
   - Write down all the steps followed for setting up the test.
   - Calculate the test statistic by hand and also code it in Python. It should be 4.76190. What decision can you make based on this calculated value?

Hint: here is the statistic [table](https://yuppal.people.ysu.edu/econ_3790/t-table.pdf ) 


###  1.- Set H0&H1 Hypothesis

H0: mu = 120

H1: mu != 120

## 2.- Calculate t-score

In [1]:
mu=120
n=100
xb=130.1
stdv = 21.21

In [2]:
import numpy as np
t_score = (xb-mu)/(stdv/np.sqrt(n))
t_score

4.761904761904759

## 3.- Compare against critical value

In [3]:
#Using scipy.stats
import scipy.stats
t_t=scipy.stats.t.ppf(q=1-.025,df=99) #from function
t_t

1.9842169515086827

In [4]:
# Value from t-table, degrees of freedom = 100-99=1
#alpha level = Confidence level = 95%

print("The critical value corresponding to a 0.95 area of a t distribution is: {:.2f}".format(t_t))

The critical value corresponding to a 0.95 area of a t distribution is: 1.98


**t > t-score:**

**Reject the null hypothesis**

### Optional
2. In a packing plant, a machine packs cartons with jars. It is supposed that a new machine will pack faster on the average than the machine currently used. To test that hypothesis, the times it takes each machine to pack ten cartons are recorded. The results, in seconds, are shown in the tables in the file `Data/machine.txt`. Assume that there is sufficient evidence to conduct the t test, does the data provide sufficient evidence to show if one machine is better than the other.

Hint: use two sample t-test

t= (sample_mean(x1) - sample_mean(x2)) / sqrt(square(s1)/n1  +  square(s2)/n2 )

In [5]:
import pandas as pd
times_df=pd.read_csv('LAb/Data/Data_Machine.txt',delimiter = "\t")
times_df.columns = ['New Machine','Old Machine']
times_df

New Machine  Old Machine
0         42.1         42.7
1         41.0         43.6
2         41.3         43.8
3         41.8         43.3
4         42.4         42.5
5         42.8         43.5
6         43.2         43.1
7         42.3         41.7
8         41.8         44.0
9         42.7         44.1

In [11]:
times_df['New Machine']

0    42.1
1    41.0
2    41.3
3    41.8
4    42.4
5    42.8
6    43.2
7    42.3
8    41.8
9    42.7
Name: New Machine, dtype: float64

In [6]:
x1= times_df['New Machine']
x2 = times_df['Old Machine']
xb1 = x1.mean()
xb2 = x2.mean()
s1 = x1.std()
s2= x2.std()
n= len(times_df)
print('New machine has an avergae speed of:' , xb1, 'with a stdev of', round(s1,3))
print('Old machine has an avergae speed of:' , round(xb2,2), 'with a stdev of', round(s2,3))

New machine has an avergae speed of: 42.14 with a stdev of 0.683
Old machine has an avergae speed of: 43.23 with a stdev of 0.75


### Equal variances

In [7]:
var1 = np.var(x1)
var2 = np.var(x2)
var2/var1

1.2038534728829622

### Seting H0 & H1

H0: xb1 - xb2 =0

H1: xb1 - xb2 <0

In [8]:
#t= (sample_mean(x1) - sample_mean(x2)) / sqrt(square(s1)/n1 + square(s2)/n2 )
t = (xb1 - xb2 )/(np.sqrt(pow(s1,2)/n + pow(s2,2)/n))
t


-3.397230706117603

In [9]:
from scipy import stats
stats.ttest_ind(x1, x2, equal_var=True)

Ttest_indResult(statistic=-3.3972307061176026, pvalue=0.0032111425007745158)

Look for t-value in table

degrees of freedom = (n1+n2) -2 = 18
Confidence level = 95%

t(.05,18) = **1.734**

 **if t stat > t-vale reject H0**
 
 -3.397 < 1.734
 
 **Fail to reject H0**